.github/workflows/beam_Release_Python_NightlySnapshot.yml

```diff
@@ -0,0 +1,92 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Release Nightly Snapshot Python
+
+on:
+  schedule:
+    - cron: '15 12 * * *'
+  workflow_dispatch:
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+  beam_Release_Python_NightlySnapshot:
+    name: ${{matrix.job_name}}  (${{matrix.job_phrase}})
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    strategy:
+      matrix:
+        job_name: [beam_Release_Python_NightlySnapshot]
+        job_phrase: [Release Nightly Snapshot Python]
+    if: github.event_name == 'workflow_dispatch' ||
+        github.event_name == 'schedule'
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup repository
+        uses: ./.github/actions/setup-action
+        with:
+          github_job: ${{matrix.job_name}}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          comment_phrase: ${{matrix.job_phrase}}
+      - name: Install Java
+        uses: actions/setup-java@v3.8.0
+        with:
+          distribution: 'zulu'
+          java-version: '8'
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: false
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Authenticate on GCP
+        uses: google-github-actions/setup-gcloud@v0
+        with:
+          service_account_email: ${{ secrets.GCP_SA_EMAIL }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          export_default_credentials: true
+      # Used because script is written with jenkins in mind
+      # TODO: Remove it once the jenkins job is disabled 
+      #       and the script is updated    
+      - name: set link
+        run: | 
```

Comment:
Am I correct in understanding that we should use the "|" approach specifically for "run," independently from the number of items to be executed? Whether it's just one or more.

Reply:
Its my preference coming from a lot of yaml. Whenever the value is not simple couple of words i use that. 

Think it adds to readability because it makes obvious the script part and not something builtin